dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/DefaultTrackerNotificationWebHookService.java

```diff
@@ -124,10 +128,19 @@ public void handleEvent( String psi )
         List<ProgramNotificationTemplate> templates = templateService
             .getProgramStageLinkedToWebHookNotifications( instance.getProgramStage() );
 
-        Map<String, String> payload = new HashMap<>();
+        // populate environment variables
+        Map<String, String> requestPayload = new HashMap<>();
         ProgramStageNotificationMessageRenderer.VARIABLE_RESOLVERS
-            .forEach( ( key, value ) -> payload.put( key.name(), value.apply( instance ) ) );
-        sendPost( templates, renderService.toJsonAsString( payload ) );
+            .forEach( ( key, value ) -> requestPayload.put( key.name(), value.apply( instance ) ) );
+
+        // populate data values
+        instance.getEventDataValues().forEach( dv -> requestPayload.put( dv.getDataElement(), dv.getValue() ) );
+
+        // populate tracked entity attributes
+        instance.getProgramInstance().getEntityInstance().getTrackedEntityAttributeValues()
```

Comment:
Do we want to add a null check or verify that this is a tracker program instance perhaps to avoid NPE?

Reply:
i think we do have a check 
https://github.com/dhis2/dhis2-core/blob/4a6b022d0246f7dc6ea841f82ffb7659352784d7/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/DefaultTrackerNotificationWebHookService.java#L97